Description: update to new API in FFmpeg 2.9, fix FTBS
Origin: upstream
Applied-Upstream: commit: r35548:r35549
--- a/src/plugins/thumbnailffmpeg_extractor.c
+++ b/src/plugins/thumbnailffmpeg_extractor.c
@@ -199,7 +199,12 @@
       return 0;
     }
 
-  if (NULL == (dst_frame = avcodec_alloc_frame ()))
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+  dst_frame = av_frame_alloc ();
+#else
+  dst_frame = avcodec_alloc_frame();
+#endif
+  if (NULL == dst_frame)
     {
 #if DEBUG
       fprintf (stderr,
@@ -215,7 +220,11 @@
       fprintf (stderr,
                "Failed to allocate the destination image buffer\n");
 #endif
-      av_free (dst_frame);
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+      av_frame_free (&dst_frame);
+#else
+      avcodec_free_frame (&dst_frame);
+#endif
       sws_freeContext (scaler_ctx);
       return 0;
     }
@@ -236,7 +245,11 @@
                "Failed to allocate the encoder output buffer\n");
 #endif
       av_free (dst_buffer);
-      av_free (dst_frame);
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+      av_frame_free (&dst_frame);
+#else
+      avcodec_free_frame (&dst_frame);
+#endif
       sws_freeContext (scaler_ctx);
       return 0;
     }
@@ -249,7 +262,11 @@
 #endif
       av_free (encoder_output_buffer);
       av_free (dst_buffer);
-      av_free (dst_frame);
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+      av_frame_free (&dst_frame);
+#else
+      avcodec_free_frame (&dst_frame);
+#endif
       sws_freeContext (scaler_ctx);
       return 0;
     }
@@ -263,10 +280,14 @@
       fprintf (stderr,
                "Failed to open the encoder\n");
 #endif
-      av_free (encoder_codec_ctx);
+      avcodec_free_context (&encoder_codec_ctx);
       av_free (encoder_output_buffer);
       av_free (dst_buffer);
-      av_free (dst_frame);
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+      av_frame_free (&dst_frame);
+#else
+      avcodec_free_frame (&dst_frame);
+#endif
       sws_freeContext  (scaler_ctx);
       return 0;
     }
@@ -295,9 +316,13 @@
 cleanup:
   av_dict_free (&opts);
   avcodec_close (encoder_codec_ctx);
-  av_free (encoder_codec_ctx);
+  avcodec_free_context (&encoder_codec_ctx);
   av_free (dst_buffer);
-  av_free (dst_frame);
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+  av_frame_free (&dst_frame);
+#else
+  avcodec_free_frame (&dst_frame);
+#endif
   sws_freeContext (scaler_ctx);
   *output_data = encoder_output_buffer;
 
@@ -406,18 +431,23 @@
       fprintf (stderr,
 	       "Failed to open image codec\n");
 #endif
-      av_free (codec_ctx);
+      avcodec_free_context (&codec_ctx);
       return;
     }
   av_dict_free (&opts);
-  if (NULL == (frame = avcodec_alloc_frame ()))
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+  frame = av_frame_alloc ();
+#else
+  frame = avcodec_alloc_frame();
+#endif
+  if (NULL == frame)
     {
 #if DEBUG
       fprintf (stderr,
                "Failed to allocate frame\n");
 #endif
       avcodec_close (codec_ctx);
-      av_free (codec_ctx);
+      avcodec_free_context (&codec_ctx);
       return;
     }
 
@@ -441,9 +471,13 @@
       fprintf (stderr,
 	       "Failed to decode a complete frame\n");
 #endif
-      av_free (frame);
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+      av_frame_free (&frame);
+#else
+      avcodec_free_frame (&frame);
+#endif
       avcodec_close (codec_ctx);
-      av_free (codec_ctx);
+      avcodec_free_context (&codec_ctx);
       return;
     }
   calculate_thumbnail_dimensions (codec_ctx->width, codec_ctx->height,
@@ -467,9 +501,13 @@
 		err);
       av_free (encoded_thumbnail);
     }
-  av_free (frame);
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+  av_frame_free (&frame);
+#else
+  avcodec_free_frame (&frame);
+#endif
   avcodec_close (codec_ctx);
-  av_free (codec_ctx);
+  avcodec_free_context (&codec_ctx);
 }
 
 
@@ -563,7 +601,12 @@
       return;
     }
 
-  if (NULL == (frame = avcodec_alloc_frame ()))
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+  frame = av_frame_alloc ();
+#else
+  frame = avcodec_alloc_frame();
+#endif
+  if (NULL == frame)
     {
 #if DEBUG
       fprintf (stderr,
@@ -616,7 +659,11 @@
       fprintf (stderr,
 	       "Failed to decode a complete frame\n");
 #endif
-      av_free (frame);
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+      av_frame_free (&frame);
+#else
+      avcodec_free_frame (&frame);
+#endif
       avcodec_close (codec_ctx);
       avformat_close_input (&format_ctx);
       av_free (io_ctx);
@@ -643,7 +690,11 @@
 		err);
       av_free (encoded_thumbnail);
     }
-  av_free (frame);
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+  av_frame_free (&frame);
+#else
+  avcodec_free_frame (&frame);
+#endif
   avcodec_close (codec_ctx);
   avformat_close_input (&format_ctx);
   av_free (io_ctx);
--- a/src/plugins/previewopus_extractor.c
+++ b/src/plugins/previewopus_extractor.c
@@ -296,7 +296,12 @@
 /** Initialize one audio frame for reading from the input file */
 static int init_input_frame(AVFrame **frame)
 {
-    if (!(*frame = avcodec_alloc_frame())) {
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+    *frame = av_frame_alloc ();
+#else
+    *frame = avcodec_alloc_frame();
+#endif
+    if (NULL == *frame) {
  #if DEBUG
         fprintf(stderr, "Could not allocate input frame\n");
 #endif
@@ -655,7 +660,11 @@
         av_freep(&converted_input_samples[0]);
         free(converted_input_samples);
     }
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+    av_frame_free (&input_frame);
+#else
     avcodec_free_frame(&input_frame);
+#endif
 
     return ret;
 }
@@ -671,7 +680,12 @@
     int error;
 
     /** Create a new frame to store the audio samples. */
-    if (!(*frame = avcodec_alloc_frame())) {
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+    *frame = av_frame_alloc ();
+#else
+    *frame = avcodec_alloc_frame();
+#endif
+    if (NULL == *frame) {
         #if DEBUG
 		fprintf(stderr, "Could not allocate output frame\n");
 		#endif
@@ -702,7 +716,11 @@
         #if DEBUG
 		fprintf(stderr, "Could allocate output frame samples (error '%s')\n", get_error_text(error));
 		#endif
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+        av_frame_free (frame);
+#else
         avcodec_free_frame(frame);
+#endif
         return error;
     }
 
@@ -783,17 +801,29 @@
         #if DEBUG
 		fprintf(stderr, "Could not read data from FIFO\n");
 		#endif
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+        av_frame_free (&output_frame);
+#else
         avcodec_free_frame(&output_frame);
+#endif
         return AVERROR_EXIT;
     }
 
     /** Encode one frame worth of audio samples. */
     if (encode_audio_frame(output_frame, output_format_context,
                            output_codec_context, &data_written)) {
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+        av_frame_free (&output_frame);
+#else
         avcodec_free_frame(&output_frame);
+#endif
         return AVERROR_EXIT;
     }
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+    av_frame_free (&output_frame);
+#else
     avcodec_free_frame(&output_frame);
+#endif
     return 0;
 }
 /** Write the trailer of the output file container. */
@@ -907,7 +937,12 @@
       return;
     }
 
-  if (NULL == (frame = avcodec_alloc_frame ()))
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55,28,1)
+  frame = av_frame_alloc ();
+#else
+  frame = avcodec_alloc_frame();
+#endif
+  if (NULL == frame)
     {
 #if DEBUG
       fprintf (stderr,
